rad/src/operators/string.rs

```diff
@@ -35,6 +36,103 @@ pub fn parse_json_array(input: &RadonString) -> Result<RadonArray, RadError> {
     item.try_into()
 }
 
+fn magic_map_cleaning(input: BTreeMap<String, RadonTypes>) -> BTreeMap<String, RadonTypes> {
+    let mut map = input;
+    let label = match map.remove("_label") {
+        Some(RadonTypes::String(rad_string)) => rad_string.value(),
+        _ => unreachable!("XML parsing error"),
+    };
+
+    match map.remove("_children") {
+        Some(RadonTypes::Array(rad_array)) => {
+            let children = rad_array.value();
+            let children_len = children.len();
+
+            match (map.is_empty(), children_len) {
+                (true, x) if x == 0 => {
```

Comment:
```suggestion
                (true, 0) => {
```

Also it would be nice to have some examples of each branch, perhaps as tests.